tests/test_patch_dc2.py

```diff
@@ -0,0 +1,161 @@
+import os
+import unittest
+from unittest.mock import patch
+
+from lsst.dc2gen3 import patch_dc2
+
+TEST_CALIB_OLD_ROOT = "/tmp/test_patch_dc2_old_calib_root"
+TEST_CALIB_NEW_ROOT = "/tmp/test_patch_dc2_new_calib_root"
+
+# Test data with which to patch os.walk,
+# so we don't need to walk a real filesystem
+TEST1_OS_WALK = [
+    (TEST_CALIB_OLD_ROOT + "/bias", ["2022-01-01"], []),
```

Comment:
I have a preference for being file system agnostic and so using `os.path.join()` for joining path segments.